Tests/MapboxNavigationTests/NavigationMapViewTests.swift

```diff
@@ -782,4 +782,78 @@ class NavigationMapViewTests: TestCase {
         
         navigationMapView.removeRoutes()
     }
+    
+    func testWaypointLayerPosition() {
+        let navigationMapView = NavigationMapView(frame: UIScreen.main.bounds)
+        
+        let styleJSONObject: [String: Any] = [
+            "version": 8,
+            "center": [
+                -122.385563, 37.763330
+            ],
+            "zoom": 15,
+            "sources": [],
+            "layers": []
+        ]
+        
+        let styleJSON: String = ValueConverter.toJson(forValue: styleJSONObject)
+        XCTAssertFalse(styleJSON.isEmpty, "ValueConverter should create valid JSON string.")
+        
+        let mapLoadingErrorExpectation = expectation(description: "Map loading error expectation")
+        mapLoadingErrorExpectation.assertForOverFulfill = false
+        
+        navigationMapView.mapView.mapboxMap.onNext(.mapLoadingError, handler: { event in
+            mapLoadingErrorExpectation.fulfill()
+        })
+        
+        navigationMapView.mapView.mapboxMap.loadStyleJSON(styleJSON)
+        
+        wait(for: [mapLoadingErrorExpectation], timeout: 5.0)
+
+        let coordinates: [CLLocationCoordinate2D] = [
+            CLLocationCoordinate2D(latitude: 37.81286886222617, longitude: -122.2784401698491),
+            CLLocationCoordinate2D(latitude: 37.81286886222617, longitude: -122.2784401698491)
+        ]
+
+        let customFixture = Fixture.route(waypoints: coordinates)
+        
+        navigationMapView.show([customFixture.route])
+        navigationMapView.showWaypoints(on: customFixture.route)
+        
+        // Style doesn't contain any layers besides the waypoint layer, main route layer, and the main route casing layer.
+        // If layer position wasn't provided, waypoint later should be placed
+        // above the main route layer.
+        // Since the waypoint layer identifier is generated by `showWaypoints(on:layerPosition:legIndex:)`,
+        // we must verify by layer type.
```

Comment:
Waypoint layers will always have `MapboxNavigation-MapboxNavigation-resources_waypointCircleLayer` or `MapboxNavigation-MapboxNavigation-resources_waypointSymbolLayer` identifiers. I think you can use `NavigationMapView.LayerIdentifier.waypointCircleLayer` and `NavigationMapView.LayerIdentifier.waypointSymbolLayer` constants in tests as well.